core-tests/src/test/java/org/mule/runtime/core/message/DefaultMuleMessageBuilderTestCase.java

```diff
@@ -272,6 +277,13 @@ public void nullPayload()
     }
 
     @Test
+    public void nullPayloadPayload()
+    {
+        MuleMessage<Object, ?> message = new DefaultMuleMessageBuilder().payload(NullPayload.getInstance()).build();
```

Comment:
Just because this is a builder test, so didn't want to depend on logic in MuleMessage.builder();